numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -540,6 +540,15 @@ PyArray_OrderConverter(PyObject *object, NPY_ORDER *val)
         return ret;
     }
     else if (!PyBytes_Check(object) || PyBytes_GET_SIZE(object) < 1) {
+        int ret = DEPRECATE("Non-string object detected for " \
```

Comment:
Maybe also just: "invalid order; only 'C', 'F', 'A', or 'K' can be converted to a memory/iteration order." (could add what was actually given, but it tends to be a bit annoying). Anyway, as I said, I am not good at writing these, just thinking loud :).